examples/standalone/model_creation/model_creation.cc

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <chrono>
+#include <iostream>
+
+#include <ignition/gazebo/components/Light.hh>
+#include <ignition/gazebo/components/LightCmd.hh>
+#include <ignition/msgs.hh>
+#include <ignition/transport.hh>
+
+using namespace std::chrono_literals;
+
+// Create a transport node.
+ignition::transport::Node node;
+
+// timeout used for services
+constexpr unsigned int timeout = 5000;
+
+void createLight()
+{
+  bool result;
+  ignition::msgs::Boolean rep;
+//! [create light]
+  ignition::msgs::EntityFactory entityFactoryRequest;
+
+  entityFactoryRequest.mutable_light()->set_name("point");
+  entityFactoryRequest.mutable_light()->set_id(22);
+  entityFactoryRequest.mutable_light()->set_parent_id(1);
```

Comment:
I think we shouldn't encourage users to hardcode IDs for their entities. Gazebo assigns the IDs when the entities are created, which is hard to predict. I think this kind of code should just use names.

IDs are helpful to uniquely identify entities at runtime more quickly, like when the server and client talk to each other.